### PR TITLE
Slow query.

### PR DESCRIPTION
**Pull request by CPWstatic**

### CODE REVIEW

File: src/meta/processors/sessionMan/SessionManagerProcessor.cpp [hunk 1]
```diff
@@ -47,6 +47,9 @@ void CreateSessionProcessor::process(const cpp2::CreateSessionReq& req) {
 void UpdateSessionsProcessor::process(const cpp2::UpdateSessionsReq& req) {
     folly::SharedMutex::WriteHolder wHolder(LockUtils::sessionLock());
     std::vector<kvstore::KV> data;
+    std::unordered_map<nebula::SessionID,
+                       std::unordered_map<nebula::ExecutionPlanID, cpp2::QueryDesc>>
```
Comment (Aiee) [inline src/meta/processors/sessionMan/SessionManagerProcessor.cpp:51]: Could use an alias here 
`using ExecPlanIdQueryDescMap = std::unordered_map<nebula::ExecutionPlanID, cpp2::QueryDesc>`

File: src/meta/processors/sessionMan/SessionManagerProcessor.cpp [hunk 2]
```diff
@@ -62,16 +65,45 @@ void UpdateSessionsProcessor::process(const cpp2::UpdateSessionsReq& req) {
             return;
         }
 
+        // update sessions to be saved if query is being killed, and return them to client.
+        auto& newQueries = *session.queries_ref();
+        std::unordered_map<nebula::ExecutionPlanID, cpp2::QueryDesc> killedQueriesInCurrentSession;
```
Comment (Aiee) [inline src/meta/processors/sessionMan/SessionManagerProcessor.cpp:70]: ditto.

File: src/meta/processors/sessionMan/SessionManagerProcessor.cpp [hunk 3]
```diff
@@ -62,16 +65,45 @@ void UpdateSessionsProcessor::process(const cpp2::UpdateSessionsReq& req) {
...
+        auto sessionInMeta = MetaServiceUtils::parseSessionVal(nebula::value(ret));
+        for (const auto& savedQuery : sessionInMeta.get_queries()) {
+            auto epId = savedQuery.first;
+            auto newQuery = newQueries.find(epId);
+            if (newQuery == newQueries.end()) {
+                continue;
+            }
+            auto& desc = savedQuery.second;
+            if (desc.get_status() == cpp2::QueryStatus::KILLING) {
+                const_cast<cpp2::QueryDesc&>(newQuery->second).set_status(desc.get_status());
```
Comment (Aiee) [inline src/meta/processors/sessionMan/SessionManagerProcessor.cpp:80]: `const_cast<cpp2::QueryDesc&>(newQuery->second).set_status(cpp2::QueryStatus::KILLING);`

**Review by Aiee — DISMISSED** [review]
LGTM.

File: src/meta/processors/sessionMan/SessionManagerProcessor.cpp [hunk 4]
```diff
@@ -145,5 +177,49 @@ void RemoveSessionProcessor::process(const cpp2::RemoveSessionReq& req) {
...
+            if (query == session.queries_ref()->end()) {
+                handleErrorCode(nebula::cpp2::ErrorCode::E_QUERY_NOT_FOUND);
+                onFinished();
+                return;
+            }
+            query->second.set_status(cpp2::QueryStatus::KILLING);
+        }
+
+        data.emplace_back(MetaServiceUtils::sessionKey(sessionId),
+                            MetaServiceUtils::sessionVal(session));
```
Comment (laura-ding) [inline src/meta/processors/sessionMan/SessionManagerProcessor.cpp:212]: alignment

File: src/meta/processors/sessionMan/SessionManagerProcessor.cpp [hunk 5]
```diff
@@ -145,5 +177,49 @@ void RemoveSessionProcessor::process(const cpp2::RemoveSessionReq& req) {
     doRemove(sessionKey);
 }
 
+void KillQueryProcessor::process(const cpp2::KillQueryReq& req) {
```
Comment (laura-ding) [inline src/meta/processors/sessionMan/SessionManagerProcessor.cpp:180]: add ci test

File: src/meta/processors/sessionMan/SessionManagerProcessor.cpp (hunk 1 above)
Comment (CPWstatic) [reply src/meta/processors/sessionMan/SessionManagerProcessor.cpp:51]: There's not much necessary to do this since we use it not that much.

**Review by laura-ding — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**